src/include/ipc/topology.h

```diff
@@ -61,9 +61,10 @@ struct sof_ipc_comp {
 	uint32_t id;
 	enum sof_comp_type type;
 	uint32_t pipeline_id;
+	uint32_t core;
```

Comment:
core id is contained within pipeline.

Reply:
Didn't he say that pipeline is too late for certain uses of this?

Reply:
> core id is contained within pipeline.

Yes, but couple of issues here:
- We cannot guarantee the order of components' creation e.g. components from pipeline id 4 will be created before pipeline itself. It keeps the flow in the FW complicated, because we need to do everything on master core: allocating buffers, notifier registrations, component get/set values etc.
- We want to have it per component anyway, since it will allow us to separate execution of particularly long high computation processing tasks to other cores.

Reply:
that is true, it does come at the end, the driver does have this information at hand so can easily insert prior to pipeline creation.